### PR TITLE
Add config rule for auto categorizing labels

### PR DESCRIPTION
**Pull request by mingshl**
To address issue #83 
Signed-off-by: Mingshi Liu <mingshl@amazon.com>

### CODE REVIEW

File: .github/draft-release-notes-config.yml [hunk 1]
```diff
@@ -17,24 +17,35 @@ categories:
...
+      - 'Security'
+      - 'security fix'
+      - 'security vulnerability'
   - title: 'Features'
     labels:
       - 'Features'
   - title: 'Enhancements'
     labels:
       - 'Enhancements'
+      - 'enhancement'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml:28]: I don't think labels should be differentiated by letter case; everything should be in lower case.

File: .github/draft-release-notes-config.yml [hunk 2]
```diff
@@ -17,24 +17,35 @@ categories:
   - title: 'Breaking Changes'
     labels:
       - 'Breaking Changes'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml]: Add `breaking`

File: .github/draft-release-notes-config.yml [hunk 3]
```diff
@@ -17,24 +17,35 @@ categories:
...
+  - title: 'Security'
+    labels:
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml:21]: add `cve` and `security feature`

File: .github/draft-release-notes-config.yml [hunk 4]
```diff
@@ -17,24 +17,35 @@ categories:
...
+      - 'Security'
+      - 'security fix'
+      - 'security vulnerability'
   - title: 'Features'
     labels:
       - 'Features'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml]: `feature`

File: .github/draft-release-notes-config.yml [hunk 5]
```diff
@@ -17,24 +17,35 @@ categories:
...
       - 'Bug Fixes'
+      - 'bug'
   - title: 'Infrastructure'
     labels:
       - 'Infrastructure'
+      - 'configuration error'
   - title: 'Documentation'
     labels:
       - 'Documentation'
+      - 'documentation'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml:37]: add `docs`

File: .github/draft-release-notes-config.yml [hunk 6]
```diff
@@ -17,24 +17,35 @@ categories:
...
   - title: 'Maintenance'
     labels:
       - 'Maintenance'
+      - 'meta'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml]: My interpretation of `meta` is a story of a series of tasks so it does not belong to category.

Comment (mingshl) [reply .github/draft-release-notes-config.yml]: removing meta

**Comment by mingshl** [comment]
@noCharger aligned to create 1 to 1 relation between label and category. modified in [label to category 1:1 mapping](https://github.com/opensearch-project/search-processor/pull/84/commits/cfeb994f9d1559537aeb64e8cd567fc1745baa6e)

File: .github/draft-release-notes-config.yml [hunk 7]
```diff
@@ -16,25 +16,28 @@ replacers:
 categories:
   - title: 'Breaking Changes'
     labels:
-      - 'Breaking Changes'
+      - 'breaking changes'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml]: 'breaking changes' -> `breaking`?

File: .github/draft-release-notes-config.yml [hunk 8]
```diff
@@ -16,25 +16,28 @@ replacers:
...
+  - title: 'Security'
+    labels:
+      - 'security fix'
   - title: 'Features'
     labels:
-      - 'Features'
+      - 'features'
```
Comment (noCharger) [inline .github/draft-release-notes-config.yml]: 'features' -> `feature`

**Review by noCharger — APPROVED** [review]
Thanks for the optimization! Only a few minor tweaks to make it more consistent.

File: .github/draft-release-notes-config.yml (hunk 8 above)
Comment (mingshl) [reply .github/draft-release-notes-config.yml]: committed in @mingshl
[singular form in label name](https://github.com/opensearch-project/search-processor/pull/84/commits/38ef698f536edf5387cf681a9b273bb4ebae0b7c)

File: .github/draft-release-notes-config.yml (hunk 7 above)
Comment (mingshl) [reply .github/draft-release-notes-config.yml]: committed in 
[singular form in label name](https://github.com/opensearch-project/search-processor/pull/84/commits/38ef698f536edf5387cf681a9b273bb4ebae0b7c)

**Comment by noCharger** [comment]
@mingshl Let's make sure all labels are created as well.

**Comment by mingshl** [comment]
> @mingshl Let's make sure all labels are created as well.

still waiting for getting maintenance access to create new labels.

**Pull request merged**